app/views/contexts/section_summary_context.py

```diff
@@ -65,16 +65,25 @@ def _build_summary(self, location):
 
     def _title_for_location(self, location):
 
-        title = self._schema.get_section(location.section_id).get("title")
-        if location.list_item_id:
-            repeating_title = self._schema.get_repeating_title_for_section(
-                location.section_id
-            )
+        section_id = location.section_id
+        list_item_id = location.list_item_id
+
+        if list_item_id:
+            repeating_title = self._schema.get_repeating_title_for_section(section_id)
             if repeating_title:
-                title = self._placeholder_renderer.render_placeholder(
-                    repeating_title, location.list_item_id
+                return self._placeholder_renderer.render_placeholder(
+                    repeating_title, list_item_id
                 )
-        return title
+
+        summary_title_for_section = self._schema.get_summary_title_for_section(
+            section_id
+        )
+        if summary_title_for_section:
+            return self._placeholder_renderer.render(
+                summary_title_for_section, list_item_id
+            )
```

Comment:
`render` expects a dict, this could be both a dict or a string, and it is just returning the string back because `find_pointers_containing(rendered_data, "placeholders")` is empty but shouldn't get to this method if it doesn't need to.